From ebbc6c1ffc2177358f7374681e72ef097d2b404f Mon Sep 17 00:00:00 2001
From: jhenrique09 <jhenrique09.mcz@hotmail.com>
Date: Mon, 23 Nov 2020 14:20:20 +0000
Subject: [PATCH 1/2] vendor: Add soong variable for
 TARGET_CAMERA_BOOTTIME_TIMESTAMP

SamarV-121: Move it under lineageGlobalVars config namespace

Signed-off-by: SamarV-121 <samarvispute121@gmail.com>
Change-Id: I5a4cfa9dfdda59b48d02b4dd5ec70c4c16511536
---
 build/soong/Android.bp     | 20 ++++++++++++++++++++
 config/BoardConfigSoong.mk |  4 +++-
 2 files changed, 23 insertions(+), 1 deletion(-)

diff --git a/build/soong/Android.bp b/build/soong/Android.bp
index b62eeb7..fe7f669 100644
--- a/build/soong/Android.bp
+++ b/build/soong/Android.bp
@@ -268,6 +268,26 @@ soong_config_module_type {
     ],
 }
 
+soong_config_module_type {
+    name: "camera_boottime_timestamp",
+    module_type: "cc_defaults",
+    config_namespace: "lineageGlobalVars",
+    bool_variables: ["needs_camera_boottime_timestamp"],
+    properties: [
+        "cflags",
+    ],
+}
+
+camera_boottime_timestamp {
+    name: "needs_camera_boottime",
+    soong_config_variables: {
+        needs_camera_boottime_timestamp: {
+            cflags: ["-DTARGET_CAMERA_BOOTTIME_TIMESTAMP"],
+        },
+    },
+}
+
+
 bootloader_message_offset {
     name: "bootloader_message_offset_defaults",
     soong_config_variables: {
diff --git a/config/BoardConfigSoong.mk b/config/BoardConfigSoong.mk
index 3c68356..5ad35b0 100644
--- a/config/BoardConfigSoong.mk
+++ b/config/BoardConfigSoong.mk
@@ -39,7 +39,8 @@ SOONG_CONFIG_lineageGlobalVars += \
     target_ld_shim_libs \
     target_process_sdk_version_override \
     target_surfaceflinger_fod_lib \
-    uses_camera_parameter_lib
+    uses_camera_parameter_lib \
+    needs_camera_boottime_timestamp
 
 SOONG_CONFIG_NAMESPACES += lineageNvidiaVars
 SOONG_CONFIG_lineageNvidiaVars += \
@@ -75,6 +76,7 @@ SOONG_CONFIG_lineageQcomVars_supports_hw_fde_perf := $(TARGET_HW_DISK_ENCRYPTION
 SOONG_CONFIG_lineageQcomVars_uses_pre_uplink_features_netmgrd := $(TARGET_USES_PRE_UPLINK_FEATURES_NETMGRD)
 SOONG_CONFIG_lineageQcomVars_uses_qcom_bsp_legacy := $(TARGET_USES_QCOM_BSP_LEGACY)
 SOONG_CONFIG_lineageQcomVars_uses_qti_camera_device := $(TARGET_USES_QTI_CAMERA_DEVICE)
+SOONG_CONFIG_lineageGlobalVars_needs_camera_boottime_timestamp := $(TARGET_CAMERA_BOOTTIME_TIMESTAMP)
 
 # Set default values
 BOOTLOADER_MESSAGE_OFFSET ?= 0
-- 
2.29.2

